experimental/dds/tree2/src/feature-libraries/editable-tree-2/proxies/types.ts

```diff
@@ -302,3 +295,23 @@ export type ProxyRoot<
 	TSchema extends TreeSchema,
 	API extends "javaScript" | "sharedTree" = "sharedTree",
 > = TSchema extends TreeSchema<infer TRootFieldSchema> ? ProxyField<TRootFieldSchema, API> : never;
+
+/** Symbol used to store a private/internal reference to the underlying editable tree node. */
+export const treeNodeSym = Symbol("TreeNode");
+
+/** Helper to retrieve the stored tree node. */
+export function getTreeNode(target: unknown): TreeNode | undefined {
+	if (typeof target === "object" && target !== null) {
+		return (target as { [treeNodeSym]?: TreeNode })[treeNodeSym];
+	}
+
+	return undefined;
+}
+
+/** Helper to set the stored tree node. */
+export function setTreeNode(target: any, treeNode: TreeNode) {
+	Object.defineProperty(target, treeNodeSym, {
+		value: treeNode,
+		configurable: true,
```

Comment:
@DLehenbauer, I had to make this "configurable: true". This was because I am now setting the property directly on the proxy rather than checking for it as a key in the `get` trap. I changed that so that `treeNodeSym` could be private to this module for organization. If this property is not configurable, then I get the "TypeError: 'ownKeys' on proxy: trap result did not include 'Symbol(TreeNode)'" error, which I think you are familiar with. 

Please advise if this is an acceptable solution, or if I should re-expose the symbol and use it in the trap. I'm not yet familiar enough with proxies to know the implications of either choice.

Reply:
This change has a side-effect of publicly exposing '[treeNodeSym]' via 'Object.getOwnPropertySymbols()'.

I believe you should be able to avoid this (and the trap error) by updating the 'ownKeys()' handler to elide '[treeNodeSym]'.

You're welcome to try the above fix or revert let me have a go at it.  (It's something I've been meaning to experiment with to simplify the 'get()' handler.)